packages/material-ui-lab/src/DatePicker/DatePicker.tsx

```diff
@@ -4,16 +4,25 @@ import useMediaQuery from '@material-ui/core/useMediaQuery';
 import PropTypes from 'prop-types';
 import DesktopDatePicker, { DesktopDatePickerProps } from '../DesktopDatePicker';
 import MobileDatePicker, { MobileDatePickerProps } from '../MobileDatePicker';
+import { ExportedCalendarPickerProps } from '../CalendarPicker/CalendarPicker';
 
 export interface DatePickerProps<TDate = unknown>
-  extends DesktopDatePickerProps<TDate>,
+  extends Omit<DesktopDatePickerProps<TDate>, 'components'>,
     MobileDatePickerProps<TDate> {
   /**
    * CSS media query when `Mobile` mode will be changed to `Desktop`.
    * @default '@media (pointer: fine)'
    * @example '@media (min-width: 720px)' or theme.breakpoints.up("sm")
    */
   desktopModeMediaQuery?: string;
+  /**
+   * The components used for each slot.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components?: ExportedCalendarPickerProps<TDate>['components'] & {
+    openPickerIcon?: React.ReactNode;
```

Comment:
It should be a component, not a node

```suggestion
    OpenPickerIcon?: React.JSXElementConstructor<any>;
```